### DISCUSSION TITLE
Multi-lingual database

### DISCUSSION THREAD

**Discussion by realJogicodes** [pocketbase/pocketbase#4950] [Q&A] [answered]
For a new app, I want to build in multi-lingual support from the get go.

My frontend is svelte kit, I can use i18n with it.

i can think of two way of using pocketbase to be multilingual.

1.  Each collection has all languages, preface the table rows with a localizer en_, es_, fr_, etc.
-> advantage: only one topical collection, sematically clean and maintainable. 
-> disadvantage: lots of "useless" data returned, overhead. Requires parsing of the return

2. Different collections for different languages. collection('en_whatever') collection('es_whatever')
-> advantage: data returned is only what is required for each language
-> disadvantage: added complexety of the database structure, more collections needed

Is there another way that I am missing?

What would you use if you had to support this?

**Comment by munxar** [comment, marked as answer]
I personally solve it with a collection that only stores the translatable fields.
You have **posts** with all non translatable fields like
- id
- author
- created
- updated

and a collection **posts_translations** with the translatable fields like

- id
- post (relation to post)
- language (relation to languages)
- title
- slug

(post,lang) should be a unique index.
and a collection **languages**

- id 
- code (de, fr, it, en or en-AT
- name (German, French, ...)

You can then query the **posts** and expand all attached translations with **expand:posts_translations_via_post** or just a specific one with a filter.

**Comment by lukepighetti** [comment]
Collection posts

id
en_title
fr_title
en_content
fr_content
